### ISSUE TITLE
Formatted Cell (pt. 2)

### ISSUE THREAD

**Issue by pzcfg** [glideapps/glide-data-grid#409]
I played around with `validateCell` and this seems to be working well! Great addition!

An additional feature that would be really nice to have is the ability to control/update the cursor position. Right now if the value is changed the cursor is forced to the end.

Here's a video showing the difference between `react-number-format`'s `NumberFormat` and rough `validateCell` implementation of it.

https://user-images.githubusercontent.com/94077014/174917652-d692dc31-d9d3-4b52-a66d-36fcf71039d4.mp4

Below is the `validateCell` function I used. It's hacky and doesn't cover all of the cases, but gets the idea across.

```
validateCell={(_cell, newValue) => {
    if (newValue.kind !== GridCellKind.Text) return true;
    const parts = `${newValue.data}`.split('.')
    const filtered = parts[0].split('').map(v => v.match(/\d/) ? v : '').join('')
    parts[0] = filtered.split('').map((v, i) => ((filtered.length - i) % 3) == 0 && i > 0 ? `,${v}` : v).join('')
    return {
        ...newValue,
        data: parts.join('.'),
    };
}}
```

**Comment by jassmith** [comment]
![Killing me](https://user-images.githubusercontent.com/30443/174931195-c1399f3a-0d24-4701-be81-24d6a4ecb15a.gif)

**Comment by pzcfg** [comment]
I know I know 🙈 I think the current `validateCell` could be extended to pass the cursor position as a second argument, and then either wrap the return value in an object with a optional `cursor` like `{ value: {}, cursor?: number } | boolean | undefined` and `{ value: { ...newValue, data: formatted }, cursor: newCursor }`, or pass in a cursor setter function like `validateCell(cell, newValue, cursor, setCursor)`, and then that gets wired up to `input.selectionStart` and `input.setSelectionRange()`

I made an example formatting hexadecimal pairs as an example here https://jsfiddle.net/d3tecpj4/

**Comment by jassmith** [comment]
I've been playing with some ideas here, I think the right approach is to return a Cell which includes a cursor postion. Basically 

```ts
type ValidateResultCell = EditableGridCell & { cursorPosition?: number };
```

If the cursor position is provided the editor can choose to respect it with a React effect. Not fully in love with this but its the only thing I could make work reliably and was not prone to a variety of timing issues.

**Comment by jassmith** [comment]
Fixed in 5.0.0
